src/System.IO.Pipelines/PipelineConnectionExtensions.cs

```diff
@@ -19,16 +19,30 @@ public static Stream GetStream(this IPipeConnection connection)
 
     public static class PipelineWriterExtensions
     {
+        private readonly static Task _completedTask = Task.FromResult(0);
+
         public static Task WriteAsync(this IPipeWriter output, Span<byte> source)
         {
             var writeBuffer = output.Alloc();
             writeBuffer.Write(source);
-            return AwaitFlushAsync(writeBuffer);
+            return FlushAsync(writeBuffer);
+        }
+
+        private static Task FlushAsync(WritableBuffer writeBuffer)
```

Comment:
There's no non generic ValueTask AFAIK. Caching the completed task here is fine.